From: Goldwyn Rodrigues <rgoldwyn@suse.com>
Subject: Correct calls to to_ib_ah_attr()
Patch-mainline: no
References: bnc#806847

The change was introduced by 4c3eb3ca13966508bcb64f39dcdef48be22f1731
where prototype of to_ib_ah_attr() was changed, however all function
calls were not converted.

Also fixed other compiler warnings wrt __devinit and vlan initialization.

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@suse.com>
[mmarek: Removed __devinit patch as it is contained in SLE11-SP3 already]
--- 
---
 drivers/infiniband/hw/mlx4/qp.c |    4 ++--
 drivers/net/mlx4/xrcd.c         |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/infiniband/hw/mlx4/qp.c
+++ b/drivers/infiniband/hw/mlx4/qp.c
@@ -2307,8 +2307,8 @@ int mlx4_ib_query_xrc_rcv_qp(struct ib_x
 
 	if (qp->ibqp.qp_type == IB_QPT_RC || qp->ibqp.qp_type == IB_QPT_UC ||
 	    qp->ibqp.qp_type == IB_QPT_XRC) {
-		to_ib_ah_attr(dev->dev, &qp_attr->ah_attr, &context.pri_path);
-		to_ib_ah_attr(dev->dev, &qp_attr->alt_ah_attr,
+ 		to_ib_ah_attr(dev, &qp_attr->ah_attr, &context.pri_path);
+ 		to_ib_ah_attr(dev, &qp_attr->alt_ah_attr,
 			      &context.alt_path);
 		qp_attr->alt_pkey_index = context.alt_path.pkey_index & 0x7f;
 		qp_attr->alt_port_num	= qp_attr->alt_ah_attr.port_num;
